internal/providers/service.go

```diff
@@ -269,6 +270,31 @@ func (_ *providerService) ValidateGitHubInstallationId(ctx context.Context, toke
 	return nil
 }
 
+// GitHubAppInstallationDeletedPayload represents the payload of a GitHub App installation deleted event
+type GitHubAppInstallationDeletedPayload struct {
+	InstallationID int64 `json:"installation_id"`
+}
+
+func (p *providerService) DeleteGitHubAppInstallation(ctx context.Context, installationID int64) error {
```

Comment:
Can we get some unit test coverage for this? It does not look like there is pre-existing coverage for this.

Reply:
Yes fair points I will add coverage. I wanted to get some feedback on the approach first - maybe I should have opened a draft to not confuse people

Reply:
That's fair - just want to make sure some more test coverage is added before merge to main.

Reply:
I opened https://github.com/stacklok/minder/pull/2885 but because there are quite a few PRs touching that logic I would prefer to unblock this PR in the meantime and work on merging the tests as we touch services.go.